### ISSUE TITLE
PROMPT_INDICATOR doesn't show up properly

### ISSUE THREAD

**Issue by fdncred** [nushell/reedline#752]
**Platform** Windows 10, MacOS, maybe others
**Terminal software** Windows Terminal, Ghostty, maybe others

Describe the problem you are observing.

When you set the PROMPT_INDOCATOR like this `$env.PROMPT_INDICATOR = {|| "\n❯ " }` the indicator doesn't show up. This was found testing this PR in nushell https://github.com/nushell/nushell/pull/11494

### Before (Note: The indicator is on the 2nd line)
![image](https://github.com/nushell/reedline/assets/343840/88019a9e-a2c5-4914-aa46-e12ef444f91c)
### After (Note: The indicator has disappeared)
![image](https://github.com/nushell/reedline/assets/343840/a500cecc-f8dc-4a34-82b0-4d993c04b66d)
### After After (Note: You can see that I typed `ls` and did not hit enter. So, the indicator is just eaten somehow)
![image](https://github.com/nushell/reedline/assets/343840/7924fdc6-0ecb-4456-b67c-1c5de7327d80)

**Comment by fdncred** [comment]
Yup. I'm gonna leave it. I thought there was an issue but couldn't find it because I was searching for PROMPT_INDICATOR. Now we have 2 issues that'll close on one fix. 😆

**Comment by kronberger-droid** [comment]
fixed in #1097
